asset: Add extrinsic documentation

vidyaa01 opened the pull request:
## Comprehensive Documentation Update

This PR introduces comprehensive documentation for the following extrinsic functions in our pallet:

- create
- issue
- transfer
- status_change
- vc_create
- vc_issue
- vc_transfer
- vc_status_change

Each function now includes detailed Rustdoc comments covering:

- Purpose: What the function does.
- Parameters: Descriptions of input parameters.
- Return Values: Information on what the function returns.
- Potential Errors: Possible errors that can be encountered.
- Events Emitted: Events triggered by the function.

Fixes: #443

vidyaa01 committed:
Update:#443

This documentation provides a comprehensive overview of the 'create' function, describing its purpose, parameters, return values, potential errors, and the event it emits upon successful execution.

vidyaa01 committed:
Merge branch 'dhiway:develop' into pallets/asset/src/lib.rs-Add-extrinsic-documentation-#443

vidyaa01 committed:
Update:#443 lib.rs 

added documentation for issue function

vidyaa01 committed:
Update lib.rs 

Added documentation for status_change function.

vidyaa01 committed:
Update lib.rs: documentation vc_create

Added documentation for vc_create function.

vidyaa01 committed:
Update lib.rs: documentation vc_issue

vidyaa01 committed:
Update lib.rs : documentation vc_transfer

vidyaa01 committed:
Update lib.rs: documentation vc_status_change

vatsa287 commented on pallets/asset/src/lib.rs:
```diff
@@ -300,6 +300,34 @@ pub mod pallet {
 
 			Ok(())
 		}
+/// Creates a new asset entry within a specified space.
```
Please move the method documentation to top of the actual method which the doc is targeted for & match its indentation.

vidyaa01 replied:
Please review it again, as I have updated the documentation accordingly.

vidyaa01 committed:
Update lib.rs: improved documentation

Added documentation above the method and corrected the indentation accordingly.

vatsa287 reviewed (dismissed):
LGTM

vatsa287 commented:
@vidyaa01 Run the cargo format for the ci test to pass.

```
cargo +nightly fmt --all
```

Add the changes again, amend the commit and push the changes.

vidyaa01 commented:
Hey @vatsa287 , 

 I have made the changes you requested and pushed the changes to the PR. It shows me dismissed stale review which wasn't my intention, can you guide me on further steps?

vatsa287 reviewed (approved):
LGTM

vatsa287 commented:
@amarts PTAL

VedantKhairnar commented:
Hello @vidyaa01   ,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.
